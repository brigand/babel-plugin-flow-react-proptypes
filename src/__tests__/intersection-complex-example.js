const babel = require('babel-core');


const content = `
var React = require('react');

import type {NamedType} from "./types";
import type DefaultType from "./types";

export type T = {
  foo: string,
}

export type U = {
  bar: number,
}

export type X = NamedType &
  { a: NamedType, b: string} &
  DefaultType & U & T;

type V = {
  qux: string,
  quy: T & NamedType & {XXX: string, YYY: NamedType & { ZZZ: number}},
  quz: U & DefaultType
} & U & T;


class C extends React.Component {
  props: V;
}

class D extends React.Component {
   props: {c: string, b: NamedType } & X;
}

export default C;
`;

it('intersection-exported-type-alias-unexported-object-literal', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
