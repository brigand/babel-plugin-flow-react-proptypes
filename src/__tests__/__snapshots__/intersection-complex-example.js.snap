exports[`test intersection-exported-type-alias-unexported-object-literal 1`] = `
"\"use strict\";

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\"Cannot call a class as a function\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\"this hasn\'t been initialised - super() hasn\'t been called\"); } return call && (typeof call === \"object\" || typeof call === \"function\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \"function\" && superClass !== null) { throw new TypeError(\"Super expression must either be null or a function, not \" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var React = require(\'react\');

var babelPluginFlowReactPropTypes_proptype_NamedType = require(\"./types\").babelPluginFlowReactPropTypes_proptype_NamedType || require(\"prop-types\").any;

var babelPluginFlowReactPropTypes_proptype_DefaultType = require(\"./types\").babelPluginFlowReactPropTypes_proptype_DefaultType || require(\"prop-types\").any;

if (typeof exports !== \"undefined\") Object.defineProperty(exports, \"babelPluginFlowReactPropTypes_proptype_T\", {
  value: {
    foo: require(\"prop-types\").string.isRequired
  }
});
if (typeof exports !== \"undefined\") Object.defineProperty(exports, \"babelPluginFlowReactPropTypes_proptype_U\", {
  value: {
    bar: require(\"prop-types\").number.isRequired
  }
});
if (typeof exports !== \"undefined\") Object.defineProperty(exports, \"babelPluginFlowReactPropTypes_proptype_X\", {
  value: Object.assign({}, babelPluginFlowReactPropTypes_proptype_NamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_NamedType, {
    a: typeof babelPluginFlowReactPropTypes_proptype_NamedType === \"function\" ? babelPluginFlowReactPropTypes_proptype_NamedType : require(\"prop-types\").shape(babelPluginFlowReactPropTypes_proptype_NamedType),
    b: require(\"prop-types\").string.isRequired
  }, babelPluginFlowReactPropTypes_proptype_DefaultType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_DefaultType, {
    bar: require(\"prop-types\").number.isRequired
  }, {
    foo: require(\"prop-types\").string.isRequired
  })
});

var C = function (_React$Component) {
  _inherits(C, _React$Component);

  function C() {
    _classCallCheck(this, C);

    return _possibleConstructorReturn(this, (C.__proto__ || Object.getPrototypeOf(C)).apply(this, arguments));
  }

  return C;
}(React.Component);

C.propTypes = {
  qux: require(\"prop-types\").string.isRequired,
  quy: require(\"prop-types\").shape(Object.assign({}, {
    foo: require(\"prop-types\").string.isRequired
  }, babelPluginFlowReactPropTypes_proptype_NamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_NamedType, {
    XXX: require(\"prop-types\").string.isRequired,
    YYY: require(\"prop-types\").shape(Object.assign({}, babelPluginFlowReactPropTypes_proptype_NamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_NamedType, {
      ZZZ: require(\"prop-types\").number.isRequired
    })).isRequired
  })).isRequired,
  quz: require(\"prop-types\").shape(Object.assign({}, {
    bar: require(\"prop-types\").number.isRequired
  }, babelPluginFlowReactPropTypes_proptype_DefaultType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_DefaultType)).isRequired,
  bar: require(\"prop-types\").number.isRequired,
  foo: require(\"prop-types\").string.isRequired
};

var D = function (_React$Component2) {
  _inherits(D, _React$Component2);

  function D() {
    _classCallCheck(this, D);

    return _possibleConstructorReturn(this, (D.__proto__ || Object.getPrototypeOf(D)).apply(this, arguments));
  }

  return D;
}(React.Component);

D.propTypes = Object.assign({}, {
  c: require(\"prop-types\").string.isRequired,
  b: typeof babelPluginFlowReactPropTypes_proptype_NamedType === \"function\" ? babelPluginFlowReactPropTypes_proptype_NamedType : require(\"prop-types\").shape(babelPluginFlowReactPropTypes_proptype_NamedType)
}, Object.assign({}, babelPluginFlowReactPropTypes_proptype_NamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_NamedType, {
  a: typeof babelPluginFlowReactPropTypes_proptype_NamedType === \"function\" ? babelPluginFlowReactPropTypes_proptype_NamedType : require(\"prop-types\").shape(babelPluginFlowReactPropTypes_proptype_NamedType),
  b: require(\"prop-types\").string.isRequired
}, babelPluginFlowReactPropTypes_proptype_DefaultType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_DefaultType, {
  bar: require(\"prop-types\").number.isRequired
}, {
  foo: require(\"prop-types\").string.isRequired
}));
exports.default = C;"
`;
